src/background/RuntimeInfoManager.ts

```diff
@@ -0,0 +1,105 @@
+/**
+ * サードパーティ向けのランタイム情報を書き出す。
+ * ランタイム情報には起動しているエンジンのURLなどが含まれる。
+ */
+
+import fs from "fs";
+import AsyncLock from "async-lock";
+import log from "electron-log/main";
+import { EngineId, EngineInfo } from "@/type/preload";
+
+/**
+ * ランタイム情報書き出しに必要なEngineInfo
+ */
+export type EngineInfoForRuntimeInfo = Pick<
+  EngineInfo,
+  "uuid" | "host" | "name"
+>;
+
+/**
+ * 保存されるランタイム情報
+ */
+type RuntimeInfo = {
+  formatVersion: number;
+  appVersion: string;
+  engineInfos: {
+    uuid: EngineId;
+    url: string;
+    name: string;
+  }[];
+};
+
+/**
+ * サードパーティ向けのランタイム情報を書き出す
+ */
+export class RuntimeInfoManager {
+  private runtimeInfoPath: string;
+  private appVersion: string;
+
+  constructor(runtimeInfoPath: string, appVersion: string) {
+    this.runtimeInfoPath = runtimeInfoPath;
+    this.appVersion = appVersion;
+  }
+
+  /**
+   * ファイルロック用のインスタンス
+   */
+  private lock = new AsyncLock({
+    timeout: 1000,
+  });
+  private lockKey = "write";
+
+  /**
+   * サードパーティ向けランタイム情報のフォーマットバージョン
+   * Note: 破壊的変更があった場合に数字を上げること
```

Comment:
テスト側で変更があったら書き換える旨は案内するようになったので省略可能
```suggestion
   * ファイルフォーマットバージョン
```